pkg/controllers/user/logging/deployer/appdeployer.go

```diff
@@ -84,10 +84,10 @@ func (d *AppDeployer) deploy(app *projectv3.App, systemWriteKeys []string) error
 		if _, err := d.AppsGetter.Apps(app.Namespace).Create(app); err != nil && !apierrors.IsAlreadyExists(err) {
 			return errors.Wrapf(err, "failed to create app %s:%s", app.Namespace, app.Name)
 		}
-	}
-
-	if current.DeletionTimestamp != nil {
-		return errors.New("stale app " + app.Namespace + ":" + app.Name + " still on terminating")
+	} else {
+		if current.DeletionTimestamp != nil {
+			return errors.New("stale app " + app.Namespace + ":" + app.Name + " still on terminating")
+		}
 	}
 
 	new := current.DeepCopy()
```

Comment:
Wouldn't this be a nil pointer too. If somehow it's getting through error check but error exists _and_ current is nil that would be case here too. Would need to either check `current != nil` or re-assign in create.